napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+
+#### Description
+This API allocates a node::Buffer object, and initializes it with data backed by the 
+passed in buffer. It exists for compat with older Node versions. Consider using typed 
+arrays instead.
+
+### *napi_create_function*
+
+#### Signature
+```
+napi_status napi_create_function(napi_env env,
+                                 const char* utf8name,
+                                 napi_callback cb,
+                                 void* data,
+                                 napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  utf8name`: A string representing the name of the function encoded as utf8
+- `[in]  cb`: A function pointer to the native function to be invoked when the created function is invoked from JavaScript
+- `[in]  data`: Optional arbitrary context data to be passed in to the callback when the function is invoked
+- `[out] result`: A `napi_value` representing a JavaScript Function
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript Function object.
+It's used to wrap native functions so that they can be invoked from JavaScript.
+
+JavaScript Functions are described in [Section 19.2](https://tc39.github.io/ecma262/#sec-function-objects) 
+of the ECMAScript Language Specification.
+
+### *napi_create_number*
+
+#### Signature
+```
+napi_status napi_create_number(napi_env env, double value, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: Double-precision value to be represented in JavaScript
+- `[out] result`: A `napi_value` representing a JavaScript Number
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to convert from C++ doubles to JavaScript Number.
+
+JavaScript Number type is described in [Section 6.1.6](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-number-type) 
+of the ECMAScript Language Specification.
+
+### *napi_create_object*
+
+#### Signature
+```
+napi_status napi_create_object(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: A `napi_value` representing a JavaScript Object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API allocates a default JavaScript Object. 
+It is the equivalent of doing `new Object()` in JavaScript.
+
+JavaScript Object type is described in [Section 6.1.7](https://tc39.github.io/ecma262/#sec-object-type) of the ECMAScript Language Specification.
+
+### *napi_create_string_utf16*
+
+#### Signature
+```
+napi_status napi_create_string_utf16(napi_env env, const char16_t* str, size_t length, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  str`: Character buffer representing a UTF16-LE encoded string
```

Comment:
If utf8 is all-lowercase, then so should UTF16-LE be. OTOH, we also need to join it to "encoded" with a hyphen, so that looks awkward: "utf16-le-encoded". So, perhaps both utf8 and utf16-le should be uppercase, so we can write "UTF8-encoded" later on, and "UTF16-LE-encoded" here.